src/charm.py

```diff
@@ -397,83 +329,35 @@ def _on_check_runners_action(self, event: ActionEvent) -> None:
-        try:
-            info = self._manager_client.check_runner()
-        except RunnerManagerServiceError as err:
-            logger.exception("Failed check runner request")
-            event.fail(f"Failed check runner request: {str(err)}")
-            return
+        info = self._manager_client.check_runner()
         event.set_results(info)
 
     @catch_action_errors
-    def _on_reconcile_runners_action(self, event: ActionEvent) -> None:
-        """Handle the action of reconcile of runner state.
-
-        Args:
-            event: Action event of reconciling the runner.
-        """
-        self.unit.status = MaintenanceStatus("Reconciling runners")
-        state = self._setup_state()
-
-        if not self._get_set_image_ready_status():
-            event.fail("Openstack image not yet provided/ready.")
-            return
-        runner_scaler = create_runner_scaler(state, self.app.name, self.unit.name)
-
-        self.unit.status = MaintenanceStatus("Reconciling runners")
-        try:
-            delta = runner_scaler.reconcile()
-        except ReconcileError:
-            logger.exception(FAILED_TO_RECONCILE_RUNNERS_MSG)
-            self.unit.status = ActiveStatus(ACTIVE_STATUS_RECONCILIATION_FAILED_MSG)
-            event.fail(FAILED_RECONCILE_ACTION_ERR_MSG)
-            return
-
-        self.unit.status = ActiveStatus()
-        event.set_results({"delta": {"virtual-machines": delta}})
-
-    @catch_action_errors
-    def _on_flush_runners_action(self, event: ActionEvent) -> None:
-        """Handle the action of flushing all runner and reconciling afterwards.
-
-        Args:
-            event: Action event of flushing all runners.
-        """
-        state = self._setup_state()
-
-        # Flushing mode not implemented for OpenStack yet.
-        runner_scaler = create_runner_scaler(state, self.app.name, self.unit.name)
-        flushed = runner_scaler.flush(flush_mode=FlushMode.FLUSH_IDLE)
-        logger.info("Flushed %s runners", flushed)
-        self.unit.status = MaintenanceStatus("Reconciling runners")
-        try:
-            delta = runner_scaler.reconcile()
-        except ReconcileError:
-            logger.exception(FAILED_TO_RECONCILE_RUNNERS_MSG)
-            self.unit.status = ActiveStatus(ACTIVE_STATUS_RECONCILIATION_FAILED_MSG)
-            event.fail(FAILED_RECONCILE_ACTION_ERR_MSG)
-            return
-        self.unit.status = ActiveStatus()
-        event.set_results({"delta": {"virtual-machines": delta}})
+    def _on_flush_runners_action(self, _: ActionEvent) -> None:
+        """Handle the action of flushing all runner and reconciling afterwards."""
+        self._manager_client.flush_runner()
 
     @catch_charm_errors
     def _on_update_status(self, _: UpdateStatusEvent) -> None:
         """Handle the update of charm status."""
-        self._ensure_reconcile_timer_is_active()
         self._log_juju_processes()
 
     def _setup_service(self, state: CharmState) -> None:
```

Comment:
will this method also be called after a unit reboot, so I guess the question would be if we can expect `config-changed` to be issued after a reboot. Otherwise we would need to set the service up on start (with a safeguard to check if image relation is already set up).

Reply:
The systemd service is set to `enable`. So systemd should start the service on machine reboot.

In the case, of only restarting the juju agent, the systemd service would not be stop in the first place.

Reply:
I will test this, just to make sure the behavior is as expected.

Reply:
From manual testing, upon machine reboot  the service does gets started.
Restarting the juju agent with systemctl does not affect the service.

Reply:
and what happens if we remove the image relation? we should disable the service then I suppose?